services/sysinfo/client/client.go

```diff
@@ -187,3 +193,159 @@ func (p *dmesgCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interfa
 	}
 	return exit
 }
+
+type journalCmd struct {
+	since  string
+	until  string
+	tail   uint64
+	unit   string
+	output string
+}
+
+func (*journalCmd) Name() string     { return "journalctl" }
+func (*journalCmd) Synopsis() string { return "Get the log entries stored in journald" }
+func (*journalCmd) Usage() string {
+	return `journalCtl [--since|--S=X] [--until|-U=X] [-tail=X] [-u|-unit=X] [-o|--output=X]9 :
+	Get the log entries stored in journald by systemd-journald.service 
+
+	Note: we have turned off the way to display all journal entries since boot since that will exceed the default buffer limit set in sansshell.
+		  Be careful when specifying the tail to a large number due to the limit of DefRunBufLimit set for stdout
+		  https://github.com/Snowflake-Labs/sansshell/blob/989cb789586532eaa22b75303ea94c97ca246306/services/util/util.go#L161.
+		  So It's best to reduce the the number of journal entries return based on these filtering flags.
+
+`
+}
+
+func (p *journalCmd) SetFlags(f *flag.FlagSet) {
+	f.StringVar(&p.since, "since", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter from")
+	f.StringVar(&p.since, "S", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter from (the date time is included)")
+	f.StringVar(&p.until, "until", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter until (the date time is not included)")
+	f.StringVar(&p.until, "U", "", "Sets the date (YYYY-MM-DD HH:MM:SS) we want to filter until")
+	f.StringVar(&p.unit, "unit", "", "Sets systemd unit to filter messages")
+	f.StringVar(&p.output, "output", "", "Sets the format of the journal entries that will be shown. Right now only json and json-pretty are supported.")
+	f.Uint64Var(&p.tail, "tail", 100, "If positive, the latest n records to fetch. By default, fetch latest 100 records.")
+}
+
+func (p *journalCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	c := pb.NewSysInfoClientProxy(state.Conn)
+
+	// the output is case insensitive
+	p.output = strings.ToLower(p.output)
+	// currently output can only be json or json-pretty
+	if p.output != "" && p.output != "json" && p.output != "json-pretty" {
```

Comment:
maybe, for the easy of use, we can just limit to one of them and they can use `jq` to easy make it pretty or compact?

Reply:
since we are going to use `jq` to transform json data into a more readable format, I will just leave output format to json. But after that, there is only one json format we need to  deal with, do we still need to leave the -output flag, I am thinking about replacing it with a boolean json flag which means `sysinfo journalctl -json `will enable the output to be json format.